### PR TITLE
850 fix for exceptions being ignored in beamline unit tests

### LINKED ISSUE #850: I22 connection tests did not fail when they weren't mocking
#847 appears to have been caused by #841 not passing `mock=True` to the `DeviceInitializationController`, that should have caused a timeout failure when the CI tried to connect to PVs that don't exist on its network, instead the tests passed after a very long time, suggesting that something was timing out but not failing the test, perhaps an exception was being swallowed.

This issue is to investigate and prevent recurrence

## Acceptance Criteria
* The cause of the non-failure is known
* A regression test/tests are in place to prevent recurrence

### PR DESCRIPTION
**Pull request by rtuck99**
Fixes #850

### Instructions to reviewer on how to test:
1. Unit tests all pass

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://diamondlightsource.github.io/dodal/main/reference/device-standards.html)
- [ ] If changing the API for a pre-existing device, ensure that any beamlines using this device have updated their Bluesky plans accordingly
- [ ] Have the connection tests for the relevant beamline(s) been run via `dodal connect ${BEAMLINE}`

### CODE REVIEW

**Review by DiamondJoseph — APPROVED** [review]
Just a nit

File: src/dodal/devices/i22/dcm.py [hunk 1]
```diff
@@ -39,21 +39,21 @@ class DoubleCrystalMonochromator(StandardReadable):
 
     def __init__(
         self,
-        motion_prefix: str,
+        prefix: str,
         temperature_prefix: str,
```
[the remark this answers is not in the document]
Comment (rtuck99) [reply src/dodal/devices/i22/dcm.py:43]: This is the downside of python having both named and positional parameters - you don't get to choose the parameter names of your functions if you want them to be called by a framework :)

[the remark this answers is not in the document]
Comment (DominicOram) [reply src/dodal/devices/i22/dcm.py:43]: I agree that we should allow not forcing `prefix` but whilst the old instantiation method insists on it I think we have to have it

File: src/dodal/beamlines/i04_1.py [hunk 2]
```diff
@@ -123,6 +123,7 @@ def undulator(
         f"{BeamlinePrefix(BL).insertion_prefix}-MO-SERVC-01:",
         wait_for_connection,
         fake_with_ophyd_sim,
+        id_gap_lookup_table_path="SPECIFY_ME",
```
Comment (DominicOram) [inline src/dodal/beamlines/i04_1.py:126]: I think we should just do https://github.com/DiamondLightSource/dodal/issues/871. I'll do that separate to this PR though

File: src/dodal/beamlines/p38.py [hunk 3]
```diff
@@ -253,6 +252,7 @@ def undulator(
         f"{BeamlinePrefix(BL).insertion_prefix}-MO-SERVC-01:",
         wait_for_connection,
         fake_with_ophyd_sim,
+        id_gap_lookup_table_path="SPECIFY_ME",
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply src/dodal/beamlines/p38.py:255]: I think all this device instantiation stuff has taught us we need a general look at how we provide these lookup tables to devices anyway. The undulator is maybe an exceptional case to this where actually in some places it's not changed, maybe we should have a writable and a read-only undulator

**Review by DominicOram — APPROVED** [review]
This is good, thanks. I agree that the `SPECIFY_ME` isn't great but I will make a new issue to holistically look at undulators

**Comment by fajinyuan** [comment]
Sorry to ask a silly question here - does P38 has undulator or insertion device?
________________________________

**Comment by DiamondJoseph** [comment]
@fajinyuan not a silly question- p38 is being set up to match the i22 bluesky experience as much as possible, so p38 has a software-only dummy undulator so that it appears in data and can be used for processing.

**Pull request merged**